### PR TITLE
Implement the reviews endpoints and models. close #359

### LINKED ISSUE #359: Implement the reviews endpoints and models
see http://dev.sphere.io/http-api-projects-reviews.html

### PR DESCRIPTION
**Pull request by schleichardt**
@lauraluiz please review, but don't merge, since I want to discuss sth. after your review which may requires additional changes.

### CODE REVIEW

File: sphere-models/src/it/java/io/sphere/sdk/reviews/commands/ReviewCreateCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                        .authorName("John Smith")
+                        .title("It is ok")
+                        .text("This product is ok.")
+                        .score(0.36)
+                        .build();
+                final Review review = execute(ReviewCreateCommand.of(reviewDraft));
+                assertThat(review.getAuthorName()).contains("John Smith");
+                assertThat(review.getTitle()).contains("It is ok");
+                assertThat(review.getText()).contains("This product is ok.");
+                assertThat(review.getScore().get()).isEqualTo(0.36, within(0.1));
```
Comment (lauraluiz) [inline sphere-models/src/it/java/io/sphere/sdk/reviews/commands/ReviewCreateCommandTest.java]: Why is it necessary the `within`? Is SPHERE rounding the score?

File: sphere-models/src/it/java/io/sphere/sdk/reviews/ReviewFixtures.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            withCustomer(client, customer -> {
+                final ReviewDraft reviewDraft = ReviewDraftBuilder.of(product, customer)
+                        .authorName("John Smith")
+                        .title("It is ok")
+                        .text("This product is ok.")
+                        .score(0.36)
+                        .build();
+                final Review review = client.execute(ReviewCreateCommand.of(reviewDraft));
+                final Review updatedReview = function.apply(review);
+                //Review delete command is not yet available in the SPHERE.IO API
```
Comment (lauraluiz) [inline sphere-models/src/it/java/io/sphere/sdk/reviews/ReviewFixtures.java]: Nice! :irony: ;)

File: sphere-models/src/it/java/io/sphere/sdk/reviews/commands/ReviewCreateCommandTest.java (hunk 1 above)
Comment (schleichardt) [inline sphere-models/src/it/java/io/sphere/sdk/reviews/commands/ReviewCreateCommandTest.java]: Yes, you cannot/should not use == with double.

**Comment by lauraluiz** [comment]
Reviewed

**Pull request merged**